src/hooks/useMerklRewardsOnCampaigns.ts

```diff
@@ -0,0 +1,190 @@
+  chainId: number;
+  address: string;
+  symbol: string;
+  accumulatedAmountForContractBN: string;
+  amountToClaim: number;
+  amountAccumulated: number;
+  proof: string[];
+  explorerLink: string;
+  chainLogo: string;
+  tokenLogo: string;
+  claimingAddress: string;
+  decimalsToShow: number;
+}
+
+export interface UseMerklRes {
+  isSuccess: boolean;
+  isLoading: boolean;
+  userTVL: number;
+  activeCampaigns: string[];
+  availableRewards: AvailableRewards[];
+  pastCampaigns: string[];
+  // activePosition?: {}[];
+}
+
+export interface UseMerklRewardsProps {
+  userAddress?: string;
+  rewardToken?: string;
+}
+
+const ACTIVE_CHAINS = REWARDS_CHAIN_IDS;
+const CREATOR_TAG = MERKL_CREATOR_TAG;
+
+const MERKL_API = 'https://api.merkl.xyz/v3';
+
+// TESTING
+// const TOKEN = '0x41A65AAE5d1C8437288d5a29B4D049897572758E';
+
+export const useMerklRewardsOnCampaigns = ({
+  userAddress,
+}: UseMerklRewardsProps): UseMerklRes => {
+  // state
+  let userTVL = 0;
+  let rewardsToClaim: AvailableRewards[] = [];
+  const activeCampaigns = [] as string[];
+  const pastCampaigns = [] as string[];
+
+  // Call to get the active positions
+  // To do -> use the label to get only
+  const MERKL_POSITIONS_API = `${MERKL_API}/multiChainPositions?chainIds=${ACTIVE_CHAINS.join(',')}&user=${userAddress}`; //&creatorTag=${CREATOR_TAG}
+  const {
+    data: positionsData,
+    isSuccess: positionsIsSuccess,
+    isLoading: positionsIsLoading,
+  } = useQuery({
+    queryKey: ['MerklPositions'],
+
+    queryFn: async () => {
+      const response = await fetch(MERKL_POSITIONS_API, {});
+      const result = await response.json();
+      return result as MerklPositionData;
+    },
+    enabled: !!userAddress,
+    refetchInterval: 1000 * 60 * 60,
+  });
+
+  // loop through the chains and positions
+  if (positionsData) {
+    for (const chain of ACTIVE_CHAINS) {
+      if (positionsData[chain]) {
+        for (const [key, data] of Object.entries(positionsData[chain])) {
+          activeCampaigns.push(key);
+        }
+      }
+    }
+  }
+
+  // check the user positions for the interesting campaign
+  const MERKL_REWARDS_API = `${MERKL_API}/rewards?chainIds=${ACTIVE_CHAINS.join(',')}&user=${userAddress}`;
+  const {
+    data: rewardsData,
+    isSuccess: rewardsIsSuccess,
+    isLoading: rewardsIsLoading,
+  } = useQuery({
+    queryKey: ['MerklRewards'],
+
+    queryFn: async () => {
+      const response = await fetch(MERKL_REWARDS_API, {});
+      const result = await response.json();
+      return result;
+    },
+    enabled: !!userAddress,
+    refetchInterval: 1000 * 60 * 60,
+  });
+
+  // transform to know what is not coming from Jumper campaigns
+
+  if (rewardsData) {
+    for (const rewardElem of REWARDS_LIST) {
+      const tokenData = rewardsData[rewardElem.chainId]?.tokenData;
+      if (tokenData) {
+        rewardsToClaim = rewardsToClaim.concat(
+          Object.entries(tokenData)
+            .map((elem): AvailableRewards => {
```

Comment:
```suggestion
            .map((elem: [number, TokenData]): AvailableRewards => {
```

is better than casting elem[1]

Reply:
thanks.